engine/robot.py

```diff
@@ -360,98 +375,131 @@ def track_obstacle(self):
                 try:
                     line = content[counter]
                     counter += 1
-                    curr_ultrasonic_value = float((''.join(line.rstrip('\n')).strip('()').split(', '))[0])
+                    curr_ultrasonic_value = float(
+                        (''.join(line.rstrip('\n')).strip('()').split(', '))[0])
                 except IndexError:
                     print("no more sensor data")
                     break
             else:
-                curr_ultrasonic_value = self.robot_state.front_ultrasonic.distance()
+                from electrical.ultrasonic_sensor import Ultrasonic
+                front_ultrasonic = Ultrasonic(0)
+                curr_ultrasonic_value = front_ultrasonic.distance()
                 if curr_ultrasonic_value < self.robot_state.front_sensor_offset:
                     self.set_phase(Phase.FAULT)
                     return None
-            if (self.robot_state.phase == Phase.TRAVERSE) or (self.robot_state.phase == Phase.RETURN) or (self.robot_state.phase == Phase.DOCKING) or (
+            if self.robot_state.control_mode == 4:  # roomba mode
```

Comment:
self.robot_state.control_mode is the control mode that the robot is in. It's something that was made before this PR. We check if its in roomba mode because if there's an obstacle in roomba_mode, we have a different obstacle avoidance mechanism to do

Reply:
I lied, I think my code was old, so we still had robot_state.control mode. Removed and updated